src/ucioption.cpp

```diff
@@ -36,6 +36,8 @@ bool CaseInsensitiveLess::operator()(const std::string& s1, const std::string& s
       [](char c1, char c2) { return std::tolower(c1) < std::tolower(c2); });
 }
 
+void OptionsMap::add_listener(OnChangeListener&& message_func) { info = std::move(message_func); }
```

Comment:
I would suggest changing `add_listener` to `add_info_listener` and `OnChangeListener` to `InfoListener` to unify it.